android/app/src/main/java/com/sphereon/ssi/wallet/MainActivity.java

```diff
@@ -71,5 +123,54 @@ public void invokeDefaultOnBackPressed() {
     super.invokeDefaultOnBackPressed();
   }
 
+  @Override
+  protected void onStop() {
+    super.onStop();
```

Comment:
Are you sure you want to call this first and not last for instance?

Reply:
After some research, I think the best practice is as follows:

Methods you override that are part of component creation (onCreate(), onStart(), onResume(), etc.), you should chain to the superclass as the first statement. This way android can do it's think before calling custom code that might depend on this.

Methods you override that are part of component destruction (onPause(), onStop(), onDestroy(), etc.), you should do your work first and chain to the superclass as the last thing. This way android cannot clean things up that the custom code might rely on

so in this case i moved the super call for the onStop and onDestroy

Reply:
this has completely been refactored